src/components/BaseButton.vue

```diff
@@ -0,0 +1,78 @@
+<template>
+    <button :class="buttonClasses" @click="$emit('click')">{{caption}}</button>
```

Comment:
It may be useful at some point to have access to original javascript event (to be able to e.g. call `e.preventDefault()`) while using this component. Because of that reason I suggest you to emit it with this event.